### ISSUE TITLE
Expounding on "opt-in for performance reasons"

### ISSUE THREAD

**Issue by jonathanhefner** [twbs/bootstrap#29378]
In the documentation for both Tooltips and Popovers is the caveat: "opt-in for performance reasons, so you must initialize them yourself."  But I'm having trouble finding a detailed explanation of "performance reasons."  The caveat is [at least 6 years old](https://github.com/twbs/bootstrap/blame/c6aae570b1bc608f5ca4b9ce9085ba9929edeb13/docs/javascript/tooltips.md#L37-L38), which makes me wonder if it's still true.

Also, many people miss this caveat when reading the documentation, for example #24985 and #15458.  So it would be great if Tooltips and Popovers could actually be automatically initialized in a performant way.  That might also pair nicely with [using `MutationObserver` to automatically remove Tooltips](https://github.com/twbs/bootstrap/issues/28964#issuecomment-507556829) in v5.

**Comment by Johann-S** [comment]
Hi @jonathanhefner, there is a way to do that, you can use our `selector` option and it'll initialize Popovers and Tooltip automatically.
Maybe after the switch to our v5 (without jQuery) we can remove that 🤔

**Comment by mdo** [comment]
Bumping @twbs/js-review on this—would we want to make this the default in v5? Is that a breaking change that we'd need to punt to v6?

**Comment by XhmikosR** [comment]
TBH we need some kind of benchmark before everything else. The reasons behind this choice may no longer apply to v5, but it can indeed be a breaking change.

**Comment by lacutah** [comment]
Follow up / tangential to this as a user of bootstrap:

1. The "opt-in" requirement and guidance already seemed weird because adding the "data-[bs-]tooltip" attribute to an element already is EXPLICIT that you want tool-tip popup behavior.
2. Inconsistency with other controls - why the additional extra step of manual JavaScript initialization when default behavior it to self-initialize from data attributes.

**Comment by mdo** [comment]
I agree with @XhmikosR on needing to re-evaluate this. @twbs/js-review Can any of you review and make a suggestion for how to better handle tooltips and popovers here without requiring extra work?

**Comment by mdo** [comment]
I think we drop this entirely and make them automatically work. Adding to v6 plans.

**Comment by Sahil3378** [comment]
v6 plan is not working here

**Comment by mdo** [comment]
Confirmed this changed in v6 for tooltips and popovers. Carousels and toasts still mention this, so need to revisit.

**Comment by mdo** [comment]
v6-dev auto-initializes tooltips and popovers via event delegation (`data-bs-toggle="tooltip"`/`"popover"`), so the manual "opt-in for performance" step no longer applies to them. Closing — please open a new issue if a specific component in v6-dev still needs this documented.
